src/handlers/walletReadyEvents.js

```diff
@@ -0,0 +1,64 @@
+import { EventEmitter } from 'events';
+import BackupStateTypes from '../helpers/backupStateTypes';
+import { Navigation } from '../navigation';
+import { addNewSubscriber } from '../redux/data';
+import store from '../redux/store';
+import { checkKeychainIntegrity } from '../redux/wallets';
+import {
+  getKeychainIntegrityState,
+  getUserBackupState,
+  saveKeychainIntegrityState,
+  saveUserBackupState,
+} from './localstorage/globalSettings';
+import Routes from '@rainbow-me/routes';
+
+const BACKUP_SHEET_DELAY_MS = 3000;
+
+export const runKeychainIntegrityChecks = () => {
+  setTimeout(async () => {
+    const keychainIntegrityState = await getKeychainIntegrityState();
+    if (!keychainIntegrityState) {
+      await store.dispatch(checkKeychainIntegrity());
+      await saveKeychainIntegrityState('done');
```

Comment:
Thinking: maybe we should saveKeychainIntegrityState('done') only after successful completion of the check in checkKeychainIntegrity (otherwise it could have failed silently inside there, without having completed the checks